lib/bolt/inventory/inventory.rb

```diff
@@ -208,17 +209,43 @@ def create_target_from_hash(data)
           add_to_group([new_target], 'all')
         end
 
-        if (aliases = new_target.target_alias)
-          aliases = [aliases] if aliases.is_a?(String)
-          unless aliases.is_a?(Array)
+        if new_target.target_alias
+          @groups.insert_alia(new_target.name, Array(new_target.target_alias))
+        end
+
+        new_target
+      end
+
+      def validate_target_from_hash(target)
+        groups = Set.new(group_names)
+        targets = target_names
+
+        # Make sure there are no group name conflicts
+        if groups.include?(target.name)
+          raise ValidationError.new("Target name #{target.name} conflicts with group of the same name", nil)
+        end
+
+        # Validate any aliases
+        if (aliases = target.target_alias)
+          unless aliases.is_a?(Array) || aliases.is_a?(String)
             msg = "Alias entry on #{t_name} must be a String or Array, not #{aliases.class}"
             raise ValidationError.new(msg, @name)
           end
-
-          @groups.insert_alia(new_target.name, aliases)
         end
 
-        new_target
+        # Make sure there are no conflicts with the new target aliases
+        aliases = @groups.target_aliases
+        target.target_alias.each do |alia|
```

Comment:
This is another place where we need to handle that `target_alias` may be a String. Since we've validated it at this point, this line could just be `Array(target.target_alias).each` since we've validated it by this point.

We should also not reuse the `aliases` variable name for two different contexts. Perhaps call the second one `used_aliases` or `known_aliases` something.